edgelet/edgelet-kube/src/lib.rs

```diff
@@ -17,11 +17,77 @@ mod registry;
 mod runtime;
 mod settings;
 
-pub use error::{Error, ErrorKind};
+pub use error::{Error, ErrorKind, Result};
 pub use module::KubeModule;
 pub use runtime::KubeModuleRuntime;
 pub use settings::Settings;
 
+use k8s_openapi::api::apps::v1 as api_apps;
+use k8s_openapi::Resource;
+use std::convert::TryFrom;
+
+type Deployment = api_apps::Deployment;
+
+#[derive(Clone)]
+pub struct KubeModuleOwner {
+    name: String,
+    api_version: String,
+    kind: String,
+    uid: String,
+}
+
+impl KubeModuleOwner {
+    pub fn new(name: String, api_version: String, kind: String, uid: String) -> Self {
+        KubeModuleOwner {
+            name,
+            api_version,
+            kind,
+            uid,
+        }
+    }
+
+    pub fn name(&self) -> &str {
+        &self.name
+    }
+
+    pub fn api_version(&self) -> &str {
+        &self.api_version
+    }
+
+    pub fn kind(&self) -> &str {
+        &self.kind
+    }
+
+    pub fn uid(&self) -> &str {
+        &self.uid
+    }
+}
+
+impl TryFrom<Deployment> for KubeModuleOwner {
+    type Error = Error;
+
+    fn try_from(deployment: Deployment) -> Result<Self> {
+        let metadata = deployment
+            .metadata
+            .as_ref()
+            .ok_or(ErrorKind::MissingMetadata)?;
+        Ok(Self {
+            name: metadata
+                .name
+                .as_ref()
+                .map(|name| name.to_string())
+                .ok_or(ErrorKind::MissingMetadata)?,
```

Comment:
now this `ErrorKind::MissingMetadata` is used in bunch of different contexts. lets make it more descriptive:
```rust
.ok_or(ErrorKind::MissingMetadata(MissingMetadataReason::Name))?
```